crates/bevy_time/src/commands.rs

```diff
@@ -0,0 +1,290 @@
+pub struct GlobalTimedCommandQueues {
+    domains_to_queues: TypeIdMap<Vec<(Timer, CommandQueue)>>,
+}
+
+impl GlobalTimedCommandQueues {
+    /// Appends the queues from the given [`TimedCommandQueues`].
+    pub fn append<Current: Domain>(&mut self, queues: &mut TimedCommandQueues<Current>) {
+        self.append_to::<Current, Current>(queues);
+    }
+
+    /// Appends the queues from the given [`TimedCommandQueues`] to the target `Target` [`Domain`]
+    /// in the [`GlobalTimedCommandQueues`].
+    ///
+    /// Queuing to a different [`Domain`] than the current one is required to support
+    /// the special-case behavior of the [`Generic`] [`Time`] domain.
+    pub fn append_to<Current: Domain, Target: Domain>(
+        &mut self,
+        queues: &mut TimedCommandQueues<Current>,
+    ) {
+        self.append_to_dyn(TypeId::of::<Target>(), queues);
+    }
+
+    /// Appends the queues from the given [`TimedCommandQueues`] to the target [`Domain`] type
+    /// in the [`GlobalTimedCommandQueues`].
+    ///
+    /// Queuing to a different [`Domain`] than the current one is required to support
+    /// the special-case behavior of the [`Generic`] [`Time`] domain.
+    pub fn append_to_dyn<Current: Domain>(
+        &mut self,
+        target_domain: TypeId,
+        queues: &mut TimedCommandQueues<Current>,
+    ) {
+        let domain = self.domains_to_queues.entry(target_domain).or_default();
+        domain.append(&mut queues.inner);
+    }
+}
+
+/// A system that ticks commands queued via [`TimedCommands`],
+/// which will be applied by default in the [`First`] and [`FixedFirst`] schedules
+/// for time domains [`Virtual`] and [`Fixed`], respectively.
+///
+/// [`First`]: bevy_app::First
+/// [`FixedFirst`]: bevy_app::FixedFirst
+/// [`Fixed`]: crate::Fixed
+pub fn queue_timed_commands<T: Domain>(
+    mut commands: Commands,
+    mut global: ResMut<GlobalTimedCommandQueues>,
+    time: Res<Time<T>>,
+) {
+    let Some(queues) = global.domains_to_queues.get_mut(&TypeId::of::<T>()) else {
+        return;
+    };
+
+    queues.retain_mut(|(timer, queue)| {
+        let finished = timer.tick(time.delta()).just_finished();
+        if finished {
+            commands.append(queue);
+        }
+        !finished
+    });
+}
+
+#[cfg(test)]
+mod tests {
+    use std::time::Instant;
+
+    use bevy_app::{App, MainSchedulePlugin, Startup, Update};
+    use bevy_ecs::{system::Resource, world::World};
+
+    use crate::TimePlugin;
+
+    use super::*;
+
+    #[test]
+    fn test() {
+        #[derive(Resource, PartialEq, Eq, Debug)]
+        struct Flag(bool);
+
+        let mut app = App::new();
+        app.add_plugins(TimePlugin);
+        app.insert_resource(Flag(false));
+        app.add_systems(Startup, |mut commands: TimedCommands| {
+            commands
+                .after(Duration::from_secs(1000))
+                .add(|world: &mut World| {
+                    *world.resource_mut::<Flag>() = Flag(true);
+                });
```

Comment:
Maybe makes sense to insert another TimedCommand using the same commands, but after the 1000 secs

```suggestion
            commands
                .after(Duration::from_secs(1000))
                .add(|world: &mut World| {
                    *world.resource_mut::<Flag>() = Flag(true);
                });
            commands
                .after(Duration::from_secs(1100))
                .add(|world: &mut World| {
                    *world.resource_mut::<Flag>() = Flag(false);
                });
```
Just so we can assert that both commands will run as expected